src/helix.idea.vim

```diff
@@ -0,0 +1,198 @@
+noremap ma[ <Esc>va[
+
+noremap mi) <Esc>vi)
+noremap mi{ <Esc>vi}
+noremap mi< <Esc>vi>
+noremap mi[ <Esc>vi]
+
+" Goto mode
+nmap gr <Action>(FindUsages)
+nmap gi <Action>(GotoImplementation)
+nmap gd <Action>(GotoDeclaration)
+nmap gT <Action>(GotoTest)
+nmap gp <Action>(PreviousTab)
+nmap gy <Action>(GotoTypeDeclaration)
+
+" Requires plugin: KJump
+nmap gw <Action>(KJumpAction.Word0)
+xmap gw <Action>(KJumpAction.Word0)
+
+" Requires plugin: String manipulation
+xmap & <Action>(StringManipulation.AlignSelections)
+
+" Selection manipulation
+nmap <A-s> <Action>(EditorAddCaretPerSelectedLine)
+vmap <A-s> <Action>(EditorAddCaretPerSelectedLine)
+nmap <A-o> <Action>(EditorSelectWord)
+xmap <A-o> <Action>(EditorSelectWord)
+nmap <A-up> <Action>(EditorSelectWord)
+xmap <A-up> <Action>(EditorSelectWord)
+nmap <A-i> <Action>(EditorUnSelectWord)
+xmap <A-i> <Action>(EditorUnSelectWord)
+nmap <A-down> <Action>(EditorUnSelectWord)
+xmap <A-down> <Action>(EditorUnSelectWord)
+nmap C <Action>(EditorCloneCaretBelow)
+xmap C <Action>(EditorCloneCaretAbove)
+
+xnoremap ~ Ugv
+nnoremap ~ vU
+xnoremap ` ugv
+nnoremap ` vu
+
+nnoremap * v"ay:<C-u>let @/='<C-r>a'<CR>
+xnoremap * "ay:<C-u>let @/='<C-r>a'<CR>
+
+" View mode
+nmap zc <Action>(EditorScrollToCenter)
+nmap zj <Action>(EditorScrollDownAndMove)
+nmap zk <Action>(EditorScrollUpAndMove)
+
+" Unimpaired
+nmap ]p <Action>(EditorForwardParagraph)
+xmap ]p <Action>(EditorForwardParagraphWithSelection)
+nmap [p <Action>(EditorBackwardParagraph)
+xmap [p <Action>(EditorBackwardParagraphWithSelection)
+
+vmap ]f <Action>(MethodDown)
+nmap ]f <Action>(MethodDown)
+xmap [f <Action>(MethodUp)
+nmap [f <Action>(MethodUp)
+xmap ]d <Action>(GotoNextError)
+nmap ]d <Action>(GotoNextError)
+xmap [d <Action>(GotoPreviousError)
+nmap [d <Action>(GotoPreviousError)
+nmap ]g <Action>(VcsShowNextChangeMarker)
+nmap [g <Action>(VcsShowPrevChangeMarker)
+
+" Space mode
+nmap <Space>f <Action>(GotoFile)
+xmap <Space>f <Action>(GotoFile)
+nmap <Space>F <Action>(FindInPath)
+xmap <Space>F <Action>(FindInPath)
+
+nmap <Space>b <Action>(RecentFiles)
+xmap <Space>b <Action>(RecentFiles)
+nmap <Space>g <Action>(RecentChangedFiles)
+xmap <Space>g <Action>(RecentChangedFiles)
+
+nmap <Space>G <Action>(Debug)
+xmap <Space>G <Action>(Debug)
+
+nmap <Space>k <Action>(QuickJavaDoc)
+xmap <Space>k <Action>(QuickJavaDoc)
+
+nmap <Space>s <Action>(FileStructurePopup)
+xmap <Space>s <Action>(FileStructurePopup)
+nmap <Space>S <Action>(GotoSymbol)
+xmap <Space>S <Action>(GotoSymbol)
+
+nmap <Space>d <Action>(ActivateProblemsViewToolWindow)
+xmap <Space>d <Action>(ActivateProblemsViewToolWindow)
+nmap <Space>D <Action>(ActivateProblemsViewToolWindow)
+xmap <Space>D <Action>(ActivateProblemsViewToolWindow)
+
+nmap <Space>c <Action>(CommentByLineComment)
+xmap <Space>c <Action>(CommentByLineComment)
+nmap <Space>C <Action>(CommentByBlockComment)
+xmap <Space>C <Action>(CommentByBlockComment)
+
+nmap <Space>y <Action>(EditorCopy)
```

Comment:
Do the registers not work in this emulator?

Reply:
Yet another thing I'll need to test again. I did do that yesterday and ended up with this, but like mentioned on another comment, I better do it with a clean `helix.vim` config.

Reply:
Are these remaps still necessary?

Reply:
Yep they are. I don't understand why though. Without these:
1. Copy text to clipboard from some other app, then paste with `<Space>p` works
2. Yank text from IDE using `<Space>y`, then do `:reg`, I see the contents are on register `"`. Paste to some other app -> nothing

Fidling with the `set` command to define registers could fix this but at least with just `helix.vim` loaded yanking does not work. `<Space>R` does not work either, but curiuosly `<Space>p` does???

Reply:
That's weird, because I defined `xnoremap <Space>y "*yg`, so I would expect it to show in the `*` register, not the `"` register. The `*` is what Vim uses as system clipboard register. Anyway, if your current workaround works, it works :)